tasks/main.yml

```
@@ -69,8 +69,22 @@
       ansible_distribution != 'Fedora' and
       ansible_distribution_major_version | int > 8
 
+- name: Ask user's consent to restart NetworkManager due to wireless or team
+    interfaces
+  fail:
+    msg: The user needs to consent to restart NetworKManager explicitly
+      by setting `network_allow_restart` to `true` when network packages
+      are changed and wireless or team connections are specified
+  when:
+    - __network_wireless_connections_defined
+       or __network_team_connections_defined
+    - network_provider == "nm"
+    - not network_allow_restart
+    - __network_package_install.changed  # noqa no-handler
```

What does package install have to do with this?

---

Restarting NM is required only when network packages are changed on the system.

---

```suggestion
    - __network_package_install is changed  # noqa no-handler
```
Ansible style prefers the `is` test - https://docs.ansible.com/ansible/2.9/user_guide/playbooks_tests.html#task-results - but if other adjacent code doesn't, then leave it as-is